pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -538,22 +562,87 @@ private static void setupTaskReceiver(Config cfg, String projectId) {
         }
     }
 
-    private static void setupFileScanResultReceiver(Config cfg, String projectId) {
+    private static void setupFileUploadListener(Config fileUploadConfig,
+            GoogleCredentials credentials,
+            String projectId,
+            PubSubConnectionManager pubsubManager) {
+        final boolean enabled = ConfigUtil.getBoolOrElse(fileUploadConfig, ConfigFile.FileUploads.ENABLE_SCAN_RESULT_HANDLER_KEY, false);
+        if (!enabled) {
+            log.info("File upload handler is disabled ({} == false)", ConfigFile.FileUploads.ENABLE_SCAN_RESULT_HANDLER_KEY);
+            return;
+        }
+
+        final var configSubscriptionNamePath = ConfigFile.FileUploads.UPLOADS_NOTIFICATION_SUBSCRIPTION_KEY;
+
+        if (!fileUploadConfig.hasPath(projectId)) {
+            log.info("A value for the key {} was not specified, file upload notifications will not be tracked.",
+                    ConfigFile.FileUploads.UPLOADS_NOTIFICATION_SUBSCRIPTION);
```

Comment:
The use of the absolute keypath here is intentional, but I have no strong ties to it. If folks feel this would improve consistency, I'm on-board for changing this to the `_KEY` suffixed value